scss/_input-group.scss

```diff
@@ -82,11 +82,11 @@
     }
   }
 
-  .btn + .btn,
-  .btn + .input-group-text,
-  .input-group-text + .input-group-text,
-  .input-group-text + .btn {
-    margin-left: -$input-border-width;
+  .input-group-text,
+  .btn {
+    &:not(:first-child) {
+      margin-left: -$input-border-width;
+    }
```

Comment:
I could also have added `.dropdown-menu` here, but figured out this would save us a lot of selectors.